src/daemon.py

```diff
@@ -0,0 +1,93 @@
+import os
+import time
+import pandas as pd
+from web3 import Web3
+from typing import List
+from threading import Thread
+from sqlalchemy import create_engine
+from sqlalchemy.engine import Engine
+from src.imbalances_script import RawTokenImbalances
+from src.config import CHAIN_RPC_ENDPOINTS, CHAIN_SLEEP_TIMES
+
+def get_web3_instance(chain_name: str) -> Web3:
+    return Web3(Web3.HTTPProvider(CHAIN_RPC_ENDPOINTS[chain_name]))
+
+def get_finalized_block_number(web3: Web3) -> int:
+    return web3.eth.block_number - 64
+
+def create_db_connection(chain_name: str):
+    """function that creates a connection to the CoW db.""" 
+    if chain_name == 'Ethereum':
+        prod_url = os.getenv("ETHEREUM_DB_URL")
```

Comment:
Not sure what my previous comment was about the use of the word `prod`, but i would suggest that we avoid it completely. The script wants to connect to the db, the environment variable does not reference "prod", and so there is no need to mention "prod" in the variable on the left-hand side.